File: c/resistor-color-duo/src/resistor_color_duo.c
#include "resistor_color_duo.h"
#include <stdio.h>

static resistor_band_t bands[] = {
  BLACK, BROWN, RED, ORANGE, YELLOW,
  GREEN, BLUE, VIOLET, GREY, WHITE
};

int color_code(resistor_band_t colors[]) {
  int color1 = colors[0];
  int color2 = colors[1];
  return (bands[color1] * 10) + bands[color2];
}

File: javascript/resistor-color-duo/resistor-color-duo.js
const RESISTOR_COLORS = [
   "black",
   "brown",
   "ted",
   "orange",
   "yellow",
   "green",
   "blue",
   "violet",
   "grey",
   "white"
]

export const decodedValue = ([color1, color2]) => {
  return (RESISTOR_COLORS.indexOf(color1) * 10) + RESISTOR_COLORS.indexOf(color2);
};
